### PR TITLE
feat: use plugin-file

### LINKED ISSUE #9: Use new file plugin

### PR DESCRIPTION
**Pull request by Julien-Torrent**
close #9

### CODE REVIEW

File: src/plugin.ts [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+      },
+    );
+
+    const copyItemTaskName = itemTaskManager.getCopyTaskName();
+    runner.setTaskPostHookHandler<Item>(
+      copyItemTaskName,
+      async (item, actor, { log = defaultLogger }, { original }) => {
+        const { id } = item; // full copy with new `id`
+
+        // TODO: check item has thumbnails
```
Comment (pyphilia) [inline src/plugin.ts]: I think we can remove this comment.

File: src/types.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    GraaspLocalFileItemOptions,
+    GraaspS3FileItemOptions,
+} from 'graasp-plugin-file';
+
+export interface GraaspThumbnailsOptions {
+    serviceMethod: ServiceMethod;
+
+    pathPrefix: string;
+
+    // TODO: use prehook in uploadPrehook.... for public
```
Comment (pyphilia) [inline src/types.ts]: Remove todo?

**Review by pyphilia — APPROVED** [review]
Nice! And so what about a prop `itemType` instead of multiple `enableFileHooks`?

File: package.json [hunk 3]
```diff
@@ -9,8 +9,8 @@
     "prepack": "yarn build",
     "prettier:check": "prettier --check \"{src,test}/**/*.ts\"",
     "prettier:write": "prettier --write \"{src,test}/**/*.ts\"",
-    "test": "jest --forceExit --coverage --verbose",
-    "test:watch": "yarn test -- --watchAll"
+    "test": "jest test --forceExit --coverage --verbose",
```
Comment (pyphilia) [inline package.json]: remove `test`?

File: src/plugin.ts [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+import {
+  THUMBNAIL_SIZES,
+  THUMBNAIL_FORMAT,
+  THUMBNAIL_PATH_PREFIX,
+  ITEM_TYPES,
+  THUMBNAIL_MIMETYPE,
+} from './utils/constants';
+import { buildFilePathWithPrefix } from './utils/helpers';
+import { AppItemExtra, GraaspThumbnailsOptions } from './types';
+import path from 'path/posix';
```
Comment (pyphilia) [inline src/plugin.ts]: import from `path` only?

File: src/plugin.ts [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+  });
+
+  if (enableItemsHooks) {
+    const deleteFileTaskName = itemTaskManager.getDeleteTaskName();
+    runner.setTaskPostHookHandler<Item>(
+      deleteFileTaskName,
+      async ({ id }, actor, { log = defaultLogger }) => {
+        //  check item has thumbnails
+        try {
+          // await access(buildFilePath(id, undefined))
```
Comment (pyphilia) [inline src/plugin.ts:132]: Can't we check that now?

File: src/plugin.ts [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+      },
+    );
+
+    const copyItemTaskName = itemTaskManager.getCopyTaskName();
+    runner.setTaskPostHookHandler<Item>(
+      copyItemTaskName,
+      async (item, actor, { log = defaultLogger }, { original }) => {
+        const { id } = item; // full copy with new `id`
+        try {
+          // copy thumbnails for copied item
```
Comment (pyphilia) [inline src/plugin.ts:154]: and check if thumbnails exist as above.
Eventually we can create a separate issue for that.

**Review by pyphilia — APPROVED** [review]

File: src/plugin.ts (hunk 6 above)
Comment (Julien-Torrent) [reply src/plugin.ts:154]: I'm gonna do it in a separate issue

**Pull request merged**